src/org/omegat/core/machinetranslators/ApertiumTranslate.java

```diff
@@ -138,4 +154,67 @@ protected String getJsonResults(String json) {
 
         return tr;
     }
+
+    private String getServerUrl() {
+        return System.getProperty(PROPERTY_APERTIUM_SERVER, Preferences.getPreference(PROPERTY_APERTIUM_SERVER));
+    }
+
+    private boolean useCustomServer() {
+        String value = System.getProperty(PROPERTY_APERTIUM_CUSTOM,
+                Preferences.getPreference(PROPERTY_APERTIUM_CUSTOM));
+        return Boolean.parseBoolean(value);
+    }
+
+    @Override
+    public boolean isConfigurable() {
+        return true;
+    }
+
+    @Override
+    public void showConfigurationUI(Window parent) {
+
+        JCheckBox apiCheckBox = new JCheckBox(OStrings.getString("APERTIUM_CUSTOM_SERVER_LABEL"));
+        apiCheckBox.setSelected(useCustomServer());
+
+        MTConfigDialog dialog = new MTConfigDialog(parent, getName()) {
+            @Override
+            protected void onConfirm() {
+                boolean temporary = panel.temporaryCheckBox.isSelected();
+                System.setProperty(PROPERTY_APERTIUM_CUSTOM, Boolean.toString(apiCheckBox.isSelected()));
+                Preferences.setPreference(PROPERTY_APERTIUM_CUSTOM, apiCheckBox.isSelected());
+                String server = panel.valueField1.getText().trim();
+                String apiKey = panel.valueField2.getText().trim();
+                System.setProperty(PROPERTY_APERTIUM_SERVER, server);
+                Preferences.setPreference(PROPERTY_APERTIUM_SERVER, server);
+                setCredential(PROPERTY_APERTIUM_API_KEY, apiKey, temporary);
+            }
+        };
+
+        apiCheckBox.addItemListener(new ItemListener() {
+            @Override
+            public void itemStateChanged(ItemEvent event) {
+                dialog.panel.valueLabel1.setEnabled(apiCheckBox.isSelected());
+                dialog.panel.valueLabel2.setEnabled(apiCheckBox.isSelected());
+                dialog.panel.valueField1.setEnabled(apiCheckBox.isSelected());
+                dialog.panel.valueField2.setEnabled(apiCheckBox.isSelected());
+                dialog.panel.temporaryCheckBox.setEnabled(apiCheckBox.isSelected());
+            }
+        });
+
+        dialog.panel.itemsPanel.add(apiCheckBox,1);
+        dialog.panel.valueLabel1.setText(OStrings.getString("APERTIUM_CUSTOM_SERVER_URL_LABEL"));
+        dialog.panel.valueField1.setText(getServerUrl());
+        dialog.panel.valueField1.setColumns(20);
+        dialog.panel.valueLabel2.setText(OStrings.getString("APERTIUM_CUSTOM_SERVER_KEY_LABEL"));
+        dialog.panel.valueField2.setText(getCredential(PROPERTY_APERTIUM_API_KEY));
+
+        dialog.panel.valueLabel1.setEnabled(apiCheckBox.isSelected());
+        dialog.panel.valueLabel2.setEnabled(apiCheckBox.isSelected());
+        dialog.panel.valueField1.setEnabled(apiCheckBox.isSelected());
+        dialog.panel.valueField2.setEnabled(apiCheckBox.isSelected());
+        dialog.panel.temporaryCheckBox.setEnabled(apiCheckBox.isSelected());
```

Comment:
You could obviate the need for these lines by storing the `ItemListener` set above as a local variable and calling its `itemStateChanged` with `null`.

Reply:
Thanks, I managed to simplify this part with your suggestion.